lib/dirList.js

```diff
@@ -94,23 +94,34 @@ const dirList = {
   /**
    * send dir list content, or 404 on error
    * @param {Fastify.Reply} reply
-   * @param {string} dir full path fs dir
+   * @param {string|string[]} dir full path fs dir
    * @param {ListOptions} options
    * @param {string} route request route
    */
   send: async function ({ reply, dir, options, route, prefix }) {
     let entries
     try {
-      entries = await dirList.list(dir, options)
+      if (Array.isArray(dir)) {
+        entries = await Promise.all(dir.map(async (d) => await dirList.list(d, options)))
```

Comment:
```suggestion
        entries = await Promise.all(dir.map((d) => dirList.list(d, options)))
```

Reply:
@Eomm 
Thanks for your comment.
I found my change would produce an inconsistency.
I created some examples.

Let's say we have some directories like this:

```
$ tree test/static3/
test/static3/
├── dir1
│   ├── hello.html
│   └── inner
│       └── index.html
└── dir2
    ├── hello.html
    ├── index.html
    ├── inner
    └── world.html

4 directories, 5 files
```

Then, the test below will pass.

```javascript
t.test('dir list adopts first found dir when the root is an array', t => {
  t.plan(2)

  const options = {
    root: [path.join(__dirname, '/static3/dir1'), path.join(__dirname, '/static3/dir2')],
    prefix: '/public',
    index: false,
    list: true
  }

  const route = '/public/'
  const content = { dirs: ['inner'], files: ['hello.html'] }

  helper.arrange(t, options, (url) => {
    t.test(route, t => {
      t.plan(3)
      simple.concat({
        method: 'GET',
        url: url + route
      }, (err, response, body) => {
        t.error(err)
        t.equal(response.statusCode, 200)
        t.equal(body.toString(), JSON.stringify(content))
      })
    })
  })
})
```

Also, the test below also passes.

```javascript
t.test('reply all files and directories when root is an array', t => {
  t.plan(2)
  const options = {
    root: [path.join(__dirname, '/static3/dir1'), path.join(__dirname, '/static3/dir2')],
    prefix: '/public',
    list: {
      format: 'json',
      names: ['index.json']
    }
  }
  const route = '/public/index.json'
  const content = { dirs: ['inner', 'inner'], files: ['hello.html', 'hello.html', 'world.html'] }

  helper.arrange(t, options, (url) => {
    t.test(route, t => {
      t.plan(3)
      simple.concat({
        method: 'GET',
        url: url + route
      }, (err, response, body) => {
        t.error(err)
        t.equal(response.statusCode, 200)
        t.equal(body.toString(), JSON.stringify(content))
      })
    })
  })
})
```

But the former test uses `/static3/dir1` only. So I'd like to update my change.